modules/browsemap/src/main/java/com/sygic/maps/module/browsemap/viewmodel/BrowseMapFragmentViewModel.kt

```diff
@@ -131,59 +132,99 @@ class BrowseMapFragmentViewModel internal constructor(
             return
         }
 
+        // Currently, we take care only of the first ViewObject
         var firstViewObject = viewObjects.first()
+        val dataLoadAllowed = onMapClickListener?.onMapClick(firstViewObject) ?: true
+
+        // First, check if the PoiDetailsView is visible
         poiDetailsView?.let {
+
+            // Always remove the previous one
             mapDataModel.removeMapObject(it)
             poiDetailsView = null
-            if (firstViewObject !is MapMarker) {
-                return
+
+            // Check if we should ask or use our own logic
+            if (onMapClickListener != null) {
```

Comment:
pozeram na tento `if-else` a nedal by sa odstranit?

na dalsom riadku je `when`. Ten:
 - ak je `MapSelectionMode.NONE` aj tak returne (neurobi nic)
- ak je `MapSelectionMode.MARKERS_ONLY` tiez checkne first aj data load allowed
- ak je `MapSelectionMode.FULL` tiez checkuje data load allowed, akuat je tam pridanie markeru, kde by sa to mohlo checknut ak je potreba

Reply:
sice asi nie, lebo to je to spravanie s tym, ze ked mame "bublinky", tak to ma iba skryt